core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2329,6 +2330,46 @@ public void multipleDiffsAgainstMultipleTablesLastFails() {
     assertThat(schema2.columns()).hasSize(1);
   }
 
+  @Test
+  public void testPaginationForListNameSpaces() {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog =
+        new RESTCatalog(SessionCatalog.SessionContext.createEmpty(), (config) -> adapter);
+    catalog.initialize("test", ImmutableMap.of(RESTSessionCatalog.REST_PAGE_SIZE, "10"));
+    int numberOfItems = 100;
+    String namespaceName = "newdb";
+
+    // create several namespaces for listing and verify
+    for (int i = 0; i < numberOfItems; i++) {
+      String nameSpaceName = namespaceName + i;
+      catalog.createNamespace(Namespace.of(nameSpaceName));
+    }
+
+    List<Namespace> results = catalog.listNamespaces();
+    assertThat(results.size()).isEqualTo(numberOfItems);
```

Comment:
```suggestion
    assertThat(results).hasSize(numberOfItems);
```

same for other assertions

Reply:
will fix